cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/RemoteDatasetFramework.java

```diff
@@ -88,7 +81,22 @@ public DatasetServiceClient load(Id.Namespace namespace) throws Exception {
         return new DatasetServiceClient(discoveryClient, namespace);
       }
     });
-    this.registryFactory = registryFactory;
+    this.instances = new LocalDatasetProvider(registryFactory) {
+      @Override
+      public DatasetMeta getMeta(Id.DatasetInstance instance) throws Exception {
+        return RemoteDatasetFramework.this.getMeta(instance);
+      }
+
+      @Override
+      public void createIfNotExists(Id.DatasetInstance instance, String type,
+                                    DatasetProperties creationProps) throws Exception {
+        if (getMeta(instance) != null) {
+          return;
+        }
+
+        RemoteDatasetFramework.this.addInstance(type, instance, creationProps);
```

Comment:
Yes, but if the instance already exists, this line won't get executed due to the previous if-statement.